### DISCUSSION TITLE
Buffer updates are a dicey affair

### DISCUSSION THREAD

**Discussion by mystilleef** [olimorris/codecompanion.nvim#507] [Q&A] [answered]
I use the `Gemini` adapter with `CodeCompanion`. I'm one of those folks that `autosaves` and `autoformats`. My apologies. 😁

So after I've accepted a diff change from a buffer update from `CodeCompanion`, `nvim` will automatically reformat and save, if no errors are present in the buffer. 

The first buffer update from `CodeCompanion` is almost always successful. But updates after that are almost always hilariously wrong. 

It seems once the buffer is reformatted, and the position of code has moved, and lines have changed, the updated state of the buffer is not reflected back to `Gemini`.  

So, unfortunately, `Gemini` is always trying to update the newly formatted buffer based on the stale information it has about the old unformated buffer.

I'm reluctant to report this as an issue because I don't know where the blame lies. Is it me, `Gemini`, or `CodeCompanion`?

Do I have to reuse the `#buffer` tag everytime the buffer changes and on every request? That doesn't seem right, but maybe I do.  Is `CodeCompanion` already doing, or not doing, this on my behalf, as I assumed? Or is `Gemini` just tripping as it tends to do? 😁

Is anyone else having this issue? If so, how did you resolve it?

**Comment by olimorris** [comment, marked as answer]
Hi @mystilleef. This is a very timely message and you're quite right on everything you've said. This is a current gap in the plugin. I'm hopefully pushing an update this weekend that allows references to be modified which is a precursor to allowing users to "pin" buffers or files to their responses (hoping that will come sometime next week). The workaround right now is to send the #buffer with every request and I acknowledge that's a nuisance.

**Reply by mystilleef** [reply]
> Hi @mystilleef. This is a very timely message and you're quite right on everything you've said. This is a current gap in the plugin. I'm hopefully pushing an update this weekend that allows references to be modified which is a precursor to allowing users to "pin" buffers or files to their responses (hoping that will come sometime next week). The workaround right now is to send the #buffer with every request and I acknowledge that's a nuisance.

@olimorris First of all, thank you for this incredible plugin. In my opinion, and trying my best not to sound hyperbolic, it's a game changer. The workaround is a minute inconvenience that pales in comparison to the incredible utility of this gem. I look forward to the updates. 👍🏿

**Reply by olimorris** [reply]
🙏🏼 Really appreciate that.

I have so many ideas in the backlog that I'm hoping I can implement over the Christmas break.
